packages/core/src/components/hotkeys/hotkeysEvents.ts

```diff
@@ -53,32 +53,50 @@ export class HotkeysEvents {
     }
 
     public handleKeyDown = (e: KeyboardEvent) => {
-        if (this.isTextInput(e) || isHotkeysDialogShowing()) {
+        const isTextInput = this.isTextInput(e);
+
+        if (isHotkeysDialogShowing()) {
             return;
         }
 
         const combo = getKeyCombo(e);
 
-        if (comboMatches(parseKeyCombo(SHOW_DIALOG_KEY), combo)) {
+        if (!isTextInput && comboMatches(parseKeyCombo(SHOW_DIALOG_KEY), combo)) {
             showHotkeysDialog(this.actions.map((action) => action.props));
             return;
         }
 
         for (const action of this.actions) {
-            if (comboMatches(action.combo, combo)) {
+            const shouldIgnore = (isTextInput && !action.props.allowInInput) || action.props.disabled;
+            if (comboMatches(action.combo, combo) && !shouldIgnore) {
+                if (action.props.preventDefault) {
+                    e.preventDefault();
+                }
+                if (action.props.stopPropagation) {
+                    e.stopPropagation();
+                }
                 safeInvoke(action.props.onKeyDown, e);
             }
         }
     }
 
     public handleKeyUp = (e: KeyboardEvent) => {
-        if (this.isTextInput(e) || isHotkeysDialogShowing()) {
+        const isTextInput = this.isTextInput(e);
+
+        if (isHotkeysDialogShowing()) {
             return;
         }
 
         const combo = getKeyCombo(e);
         for (const action of this.actions) {
-            if (comboMatches(action.combo, combo)) {
+            const shouldIgnore = (isTextInput && !action.props.allowInInput) || action.props.disabled;
+            if (comboMatches(action.combo, combo) && !shouldIgnore) {
+                if (action.props.preventDefault) {
+                    e.preventDefault();
+                }
+                if (action.props.stopPropagation) {
+                    e.stopPropagation();
+                }
```

Comment:
actually, i think the only difference between the handleKeyDown/Up is which function gets `safeInvoke`d, so perhaps we can refactor even harder?